Lobby and game now supports playing vs AI

sjohan99 opened the pull request:
**Merge this after [this PR](https://github.com/ChessVariants/ChessVariantsAPI/pull/26) as that contains the backend functionality required.**

Pretty basic stuff! I've added a method in `GameService` for sending a request to the backend to add an AI to the game. Additionally the lobby now has a "Add AI" button which is available only when the lobby is not full.

This PR should be merged after https://github.com/ChessVariants/chess-variants-frontend/pull/7 since that PR changes the lobby somewhat, so I'll adjust these changes accordingly.

ValleSten commented on src/Components/SetupGame/Lobby/Lobby.tsx:
```diff
@@ -38,6 +40,25 @@ export default function Lobby(props: { gameID: string, isAdmin: boolean }) {
+            }
+        })
+
+        gameService.on(GameEvents.PlayerJoinedGame, (color: string, name: string) => {
+            if (name != username) {
+                setLobbyFull(true)
+            }
+        })
```
The parameter _color_ is never used, is it necessary to have it here? If you were to remove it, the corresponding PR in the back end of course needs to be updated as well.

ValleSten reviewed (changes requested):
Looks good, I have only left two identical comments that I'd like some response to. There's also a small merge conflict that needs to be addressed, I suppose it's following the merging of PR #7 that you referred to.

sjohan99 commented:
Merge conflics fixed and I responded to one of the comments. Idk why the code was duplicated there but I removed one of them and changed the other one to use an underscore instead of `color` to indicate that it isn't being used.

ValleSten reviewed (approved):
Ah, I see. This looks great. Feel free to merge whenever you're ready :)

ValleSten commented on src/Components/SetupGame/Lobby/Lobby.tsx:
```diff
@@ -38,6 +40,25 @@ export default function Lobby(props: { gameID: string, isAdmin: boolean }) {
+            if (name != username) {
+                setLobbyFull(true)
+            }
+        })
```
Ah I see!
